### PR TITLE
Home Work 9

### PR DESCRIPTION
**Pull request by TanyaB777**

### CODE REVIEW

File: src/main/java/hw_9/task1/Planet.java [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package hw_9.task1;
+
+public interface Planet {
+
+    final String fromSun = " from the SUN: ";
```
Comment (AntBelyi) [inline src/main/java/hw_9/task1/Planet.java]: "final" key word can be removed

File: src/main/java/hw_9/task1/Planet.java [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+    default void displayInfo() {
+        System.out.println(this.getClass().toString());
+    }
+
+    ;
```
Comment (AntBelyi) [inline src/main/java/hw_9/task1/Planet.java]: this symbol can be removed?

File: src/main/java/hw_9/task1/PlanetFactory.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+            case "Jupiter":
+                return new Jupiter();
+            case "Saturn":
+                return new Saturn();
+            case "Uranus":
+                return new Uranus();
+            case "Neptune":
+                return new Neptune();
+            default:
+                return null;
```
Comment (AntBelyi) [inline src/main/java/hw_9/task1/PlanetFactory.java]: this return type is bad. Try to implement an Exception

File: src/main/java/hw_9/task1/Planet.java [hunk 4]
```diff
@@ -0,0 +1,12 @@
+package hw_9.task1;
+
+public interface Planet {
+
+    final String fromSun = " from the SUN: ";
+
+    default void displayInfo() {
+        System.out.println(this.getClass().toString());
```
Comment (AntBelyi) [inline src/main/java/hw_9/task1/Planet.java]: this sout display not very useful information
Screenshot_6

So, i guess we don't need to do it as default, just - void displayInfo();

File: src/main/java/hw_9/task1/Earth.java [hunk 5]
```diff
@@ -0,0 +1,11 @@
+package hw_9.task1;
+
+public class Earth implements Planet {
+    String name = "Earth";
```
Comment (AntBelyi) [inline src/main/java/hw_9/task1/Earth.java]: where is access modifier? And probably it is a constant? Fix in all places please

File: src/main/java/hw_9/task1/Planet.java (hunk 1 above)
Comment (TanyaB777) [reply src/main/java/hw_9/task1/Planet.java]: fixed

File: src/main/java/hw_9/task1/Planet.java (hunk 2 above)
Comment (TanyaB777) [reply src/main/java/hw_9/task1/Planet.java]: fixed

File: src/main/java/hw_9/task1/PlanetFactory.java (hunk 3 above)
Comment (TanyaB777) [reply src/main/java/hw_9/task1/PlanetFactory.java]: fixed

File: src/main/java/hw_9/task1/Planet.java (hunk 4 above)
Comment (TanyaB777) [reply src/main/java/hw_9/task1/Planet.java]: fixed

File: src/main/java/hw_9/task1/Earth.java (hunk 5 above)
Comment (TanyaB777) [reply src/main/java/hw_9/task1/Earth.java]: fixed

File: src/main/java/hw_9/task1/Earth.java [hunk 6]
```diff
@@ -0,0 +1,11 @@
+package hw_9.task1;
+
+public class Earth implements Planet {
+    private final String name = "Earth";
```
Comment (AntBelyi) [inline src/main/java/hw_9/task1/Earth.java]: 1) constants should be in upper case - NAME
2) generally constants are static - private static final String <name> = ....

fix in all places please

File: src/main/java/hw_9/task1/Program.java [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+
+        String[] planetNames = {"Mercury", "Venus", "Earth", "Mars", "Jupiter", "Saturn", "Uranus", "Neptune"};
+
+        try {
+            for (String helperPlanet : planetNames) {
+                PlanetFactory.getPlanet(helperPlanet).displayInfo();
+            }
+        } catch (Exception e) {
+            System.out.println(e.getMessage());
+        }
```
Comment (AntBelyi) [inline src/main/java/hw_9/task1/Program.java:10]: try catch can be removed. Leave a cycle only

File: src/main/java/hw_9/task1/PlanetFactory.java [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+            case "Uranus":
+                return new Uranus();
+            case "Neptune":
+                return new Neptune();
+            default:
+                try {
+                    throw new Exception("Planet with this name is not exist!");
+                } catch (Exception e) {
+                    throw new RuntimeException(e);
+                }
```
Comment (AntBelyi) [inline src/main/java/hw_9/task1/PlanetFactory.java]: Could you fix using this way?
image

File: src/main/java/hw_9/task1/Earth.java (hunk 6 above)
Comment (TanyaB777) [reply src/main/java/hw_9/task1/Earth.java]: fixed

File: src/main/java/hw_9/task1/Program.java (hunk 7 above)
Comment (TanyaB777) [reply src/main/java/hw_9/task1/Program.java:10]: fixed

File: src/main/java/hw_9/task1/PlanetFactory.java (hunk 8 above)
Comment (TanyaB777) [reply src/main/java/hw_9/task1/PlanetFactory.java]: fixed

**Pull request merged**